### ISSUE TITLE
dojo path is not correct on client.html on version 2.2.3

### ISSUE THREAD

**Issue by alvaroortega** [theintern/intern#871]
The paths to dojo dependency seems to point to the wrong location on intern 2.2.3. 
When this version is installed using npm, dojo is installed under `/node_modules/@theintern/dojo` however on client.html paths are defined like:

```
intern: {
    dojo: 'intern/node_modules/dojo',
    chai: 'intern/node_modules/chai/chai',
    diff: 'intern/node_modules/diff/diff'
},
'*': {
    'intern/dojo': 'intern/node_modules/dojo'
}
```
When I try to run intern I got 404.

Intern version: 2.2.3
Node version: v4.4.4
NPM version: 2.15.1

I have solved it temporarily by modifying client.html

````
internConfig = this.__internConfig = {
        // For users running client.html directly, assume that the base path
        // is two levels up (the parent of node_modules); this is most common.
	// Users that need something special should provide an absolute module
	// path for the config module (e.g. `/path/to/intern/config`) and
	// then set `baseUrl` correctly in the `loader` configuration object
	// of the config module
	baseUrl: basePath.slice(-10) === '/__intern/' ? basePath.slice(0, -9) : (basePath + '../../'),
	packages: [
		{ name: 'intern', location: basePath }
	],
	map: {
		intern: {
			dojo: 'intern/node_modules/@theintern/dojo',
			chai: 'intern/node_modules/chai/chai',
			diff: 'intern/node_modules/diff/diff'
		},
		'*': {
			'intern/dojo': 'intern/node_modules/@theintern/dojo'
		}
	}
};
```

**Comment by jason0x43** [comment]
When intern is installed, dojo will be symlinked from its original location to `node_modules/intern/dojo`. In the past this was `node_modules/dojo`, now it's `node_modules/@theintern/dojo`. 

Unfortunately, the fixdeps logic assumes that `@theintern/dojo` was hoisted up to the main `node_modules` directory, which Node 4 apparently isn't doing. _sigh_

**Comment by alvaroortega** [comment]
In case it is of any help. I just found a way to solve this in our side which is using the config file to set the map property.

```
loader: {
...
        map: {
            intern: {
                dojo: 'intern/node_modules/@theintern/dojo',
                chai: 'intern/node_modules/chai/chai',
                diff: 'intern/node_modules/diff/diff'
            },
            '*': {
                'intern/dojo': 'intern/node_modules/@theintern/dojo'
            }
...
        },
```

**Comment by alvaroortega** [comment]
Also, you are right. newer versions of node are hoisting the deeper dependencies to the main node_modules but this would break a big chunk of legacy code for us.

**Comment by alvaroortega** [comment]
I correct myself. I think the cache played on me. That fix I mentioned does not really have any effect. I need a way to update the path from outside the client.html

**Comment by jason0x43** [comment]
#351 was never merged, but even without that, you can't override the config used in `client.html` (at least not at the point Intern is loaded) without modifying `client.html`.

The two fixes for this problem are to 1) fix the `fixdeps.js` script to link dojo properly, or 2) use the link currently created by `fixdeps.js` and update all the references to 'node_modules/dojo' to use 'node_modules/@theintern/dojo'. I'd prefer to do the former, because it involves fewer modifications to the source, but I haven't had a chance to test it yet.

**Comment by alvaroortega** [comment]
Ahh, sorry I thought it was. Indeed I realised client.html is parsed by the browser before intern is taken into account so I abandoned that possibility. I thought I could use a custom runner task on grunt to parse the file and modify the string as changing node version is breaking loads of other things.

I'm trying in parallel upgrading to version 3 but it also requires changing the config a bit and probably rewrite reporter according to release notes.

Just out of curiosity, how happen the path had to be changed?

Thanks

**Comment by jason0x43** [comment]
Intern 2 had been using a custom version of dojo hosted in a developer's personal GitHub account. This version of dojo called itself "dojo" in its package.json, so that's where npm would install it (`node_modules/dojo` or `node_modules/intern/node_modules/dojo`). That repository was recently removed, so we made a new copy of it and published it to the npm repo (to prevent this sort of issue in the future). Since this dojo isn't the real dojo, we didn't want to publish it as "dojo" in the npm registry, so it was published as `@theintern/dojo`.

**Comment by alvaroortega** [comment]
That makes perfect sense

**Comment by jason0x43** [comment]
Fixed in f8ad1c1
